test/Riok.Mapperly.IntegrationTests/_snapshots/NET_48/StaticMapperTest.SnapshotGeneratedSource.verified.cs

```diff
@@ -374,6 +375,26 @@ public static partial class StaticTestMapper
             };
         }
 
+        public static partial TTarget MapGeneric<TSource, TTarget>(TSource source)
+        {
+            return source switch
+            {
+                global::Riok.Mapperly.IntegrationTests.Models.TestEnum x when typeof(TTarget).IsAssignableFrom(typeof(global::Riok.Mapperly.IntegrationTests.Dto.TestEnumDtoByName)) => (TTarget)(object)MapToEnumDtoByName(x),
```

Comment:
Is the extra cast to `object` needed before casting it to `TTarget`?

Reply:
Yep it is 🤷